docs/routing-logic.md

```diff
@@ -0,0 +1,114 @@
+**Trino Gateway documentation**
+
+<table>
+  <tr>
+    <td>
+      <img src="./assets/logos/trino-gateway-v.png"/>
+    </td>
+    <td>
+      <ul>
+        <li><a href="quickstart.md">Quickstart</a></li>
+        <li><a href="installation.md">Installation</a></li>
+        <li><a href="security.md">Security</a></li>
+        <li><a href="operation.md">Operation</a></li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li><a href="gateway-api.md">Gateway API</a></li>
+        <li><a href="resource-groups-api.md">Resource groups API</a></li>
+        <li><a href="routing-rules.md">Routing rules</a></li>
+        <li><a href="routing-logic.md">Routing logic</a></li>
+      </ul>
+    </td>
+    <td>
+      <ul>
+        <li><a href="design.md">Design</a></li>
+        <li><a href="development.md">Development</a></li>
+        <li><a href="release-notes.md">Release notes</a></li>
+        <li><a href="references.md">References</a></li>
+      </ul>
+    </td>
+  </tr>
+</table>
+
+# Routing Logic
+
+## Overview
+Incoming requests are evaluated by the Gateway to determine if they
+are a follow-up to a request previously handled by the Gateway. If the 
+Gateway determines that they are, they are forwarded to whatever the 
+backend that handled the previous request.
+
+If the request does not have an antecedent, the Gateway first uses [Routing Rules](routing-rules.md)
+to determine which logical group of clusters, termed a "Routing Group",
+should handle the request. A member of the Routing Group, termed a backend, 
+is selected to handle the request using either an adaptive or round-robin 
+strategy.
+
+![Request Routing Flow](assets/gateway-routing-flow.svg)
```

Comment:
Can you add a start/user in the flow? I assume it's "is request descended from a previous request"

Reply:
Not sure why I'm unable to respond to these individually

> Why would you need user to configure this? Other than logout?
There is no generic reason. However, the cookie routing might be leveraged in use case specific ways. For example, maybe some workflow using [Dune's table API](https://docs.dune.com/api-reference/tables/endpoint/create) would benefit from pinning to a specific cluster, then unpinning after a certain request.

Are you looking for more specific guidance on this in the docs?

> In which cases would you need custom rules?
Again, no generic reason, but there could be a use case specific reason. For example, Starburst has a few different APIs for managing access control, caching and data virtualization. There may be some case where you want to unpin depending on something in the response instead of just the path. However, this is out of scope for the current work so I've removed the TODO